test/e2e/util/hypershift_framework.go

```diff
@@ -322,3 +328,22 @@ func (h *hypershiftTest) teardownHostedCluster(ctx context.Context, hc *hyperv1.
 		}
 	}
 }
+
+func (h *hypershiftTest) summarizeHCConditions(hostedCluster *hyperv1.HostedCluster) {
+	conditions := hostedCluster.Status.Conditions
+
+	if conditions != nil {
+		h.Logf("Summarizing false conditions for HostedCluster %s ", hostedCluster.Name)
+		for _, condition := range conditions {
+			if condition.Status == metav1.ConditionFalse {
```

Comment:
One final comment - @enxebre mentioned that it might not always be the case that conditions are "good" when they are `condition.Status == metav1.ConditionTrue` - I certainly know of a number of examples in the wild where the correct status is false. 

Do we want to quickly commit a `map[string]metav1.ConditionStatus` to record - for the known ones - if we expect them to be `true` or `false`? We can either always log ones we don't know about or only log unknown ones when false.

Reply:
Starting to think we should just rework and reuse https://github.com/openshift/hypershift/blob/main/test/e2e/util/util.go#L1451. wdyt ?

Reply:
Oh wow - I am new to this repo so I did not know that existed. It looks good - I would add in your logic from this PR to emit the message & reason, though, since the one you linked seems to miss that. Maybe @csrwng can chime in?

Reply:
The reason behind this pr is to make the conditions more visible on any test failure. The one linked above is ran before the main tests to check that the cluster is in a good state. From my understanding that test could pass because at the time all the conditions were "good" but that doesnt mean that the conditions wont change throughout the duration of the e2e tests. 

I think a rework of  https://github.com/openshift/hypershift/blob/main/test/e2e/util/util.go#L1451 to include some logs could be worth it. We  then can run it before main tests and again on failures. 

wdyt @enxebre

Reply:
The main issue as stated in the jira is that if a HC doesn't complete rollout i.e. when WaitForImageRollout fails we log a bunch of noisy stuff and don't provide any clear reason.
When I open `TestCreateClusterProxy/ValidateHostedCluster` [here](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-storage-operator/403/pull-ci-openshift-cluster-storage-operator-master-hypershift-aws-e2e-external/1709056236764270592) I should see some pointers to why rollout doesn't completed.
I'd start by removing the verbosity from WaitForImageRollout and I'd integrate their logic with the one in validateHostedClusterConditions so when WaitForImageRollout logs conditions in unexpected state.
But as far as the above outcome is achieve I don't have strong opinions on the implementation here.